python/private/attributes.bzl

```diff
@@ -253,25 +242,24 @@ def _create_native_rules_allowlist_attrs():
 NATIVE_RULES_ALLOWLIST_ATTRS = _create_native_rules_allowlist_attrs()
 
 # Attributes common to all rules.
-COMMON_ATTRS = union_attrs(
+COMMON_ATTRS = dicts.add(
     DATA_ATTRS,
     NATIVE_RULES_ALLOWLIST_ATTRS,
     # buildifier: disable=attr-licenses
     {
         # NOTE: This attribute is deprecated and slated for removal.
-        "distribs": attr.string_list(),
+        ##"distribs": attr.string_list(),
         # TODO(b/148103851): This attribute is deprecated and slated for
         # removal.
         # NOTE: The license attribute is missing in some Java integration tests,
         # so fallback to a regular string_list for that case.
         # buildifier: disable=attr-license
-        "licenses": attr.license() if hasattr(attr, "license") else attr.string_list(),
+        ##"licenses": attr.license() if hasattr(attr, "license") else attr.string_list(),
     },
-    allow_none = True,
 )
 
 IMPORTS_ATTRS = {
-    "imports": attr.string_list(
+    "imports": lambda: rule_builders.StringListAttrBuilder(
```

Comment:
I like this definition of attrs. Having the values as lambdas feels somewhat intuitive.